classes/cliTool/CommandLineTool.php

```diff
@@ -67,7 +67,11 @@ public function __construct($argv = [])
         $request->setRouter($router);
 
         // Initialize the locale and load generic plugins.
-        PluginRegistry::loadCategory('generic');
+        $contextDao = Application::getContextDAO(); /** @var ContextDAO $contextDao */
+        $contextFactory = $contextDao->getAll(); /** @var DAOResultFactory $contextFactory */
+        while ($context = $contextFactory->next()) { /** @var Context $context */
+            PluginRegistry::loadCategory('generic', false, $context->getId());
```

Comment:
Does it this cover site-wide plugins (I didn't inspect)?
Another possible issue is running a CLI tool when there's no contexts (is there a use-case for that?)

Perhaps it's ok to keep the previous `PluginRegistry::loadCategory('generic');` together with this new code (I remember there was something to avoid registering the same plugin twice).

Reply:
I think it does not matter for side-wide plugins -- they do nothing with mainContextId -- they register their hooks in any case and do not check if they are enable for that context. 
And regarding loading plugins: 
mainContextId matters here:
https://github.com/pkp/pkp-lib/blob/main/classes/plugins/PluginRegistry.php#L112
then here:
https://github.com/pkp/pkp-lib/blob/main/classes/plugins/PluginRegistry.php#L216
then here:
https://github.com/pkp/pkp-lib/blob/main/classes/core/PKPApplication.php#L427
i.e. https://github.com/pkp/pkp-lib/blob/main/classes/site/VersionDAO.php#L190
all plugins, also side-wide will be considered.
But, I am not 100% sure, maybe @asmecher will know better.

Reply:
I'll double check the plugins regarding `$this->context = $journalDao->getById($mainContextId);` but I don't think so -- it could cause a problem earlier too...